[Week3 27조 Android] 3주차 산출물 제출합니다.

fiveJinw opened the pull request:
## 📋 작업 개요
템플릿, Pull Request 템플릿 생성
정보 등록 flow 화면 xml 구성 및 bottom navigation 구현(기능 제외)

## 📝 작업 상세 설명
깃허브 이슈 생성시 사용할 템플릿과 Pull Request시 자동으로 생성되는 템플릿을 작성했습니다.
이슈 라벨은 총 11개로 각각 Status, Type, Priority를 기반으로 구분해서 사용하도록 했습니다.
개발을 진행하면서 사용할 color 및 초기 drawable 파일들을 등록하였습니다.
bottom navigation을 초기 기획에 맞게끔 구성하였습니다.
사용자의 정보를 등록할 수 있는 화면의 xml을 구성하였습니다.

### ✅ 체크리스트
- [x] 모든 테스트를 통과했나요?
- [x] 커밋 컨벤션에 맞춰서 작성 했나요?
- [x] PR을 요청할 브랜치를 제대로 선택했나요?
- [x] 이전의 업데이트 사항을 전부 반영했나요?

### 🔗 관련 이슈
없습니다.

### 📚 참고 사항
[27조 프로젝트](https://github.com/orgs/kakao-tech-campus-2nd-step3/projects/20/views/1)
[3주차 위클리 프로젝트 보고서](https://www.notion.so/2381b2d578374384bb23d6764b540969?p=24a1075b1df94a4da1f128a8add6d2b0&pm=s)

fiveJinw committed:
init : 초기 설정

fiveJinw committed:
chore : 글꼴 추가

yb0x00 committed:
Merge pull request #1 from fiveJinw/Week3

init : 초기 설정

fiveJinw committed:
[Week3] 프로젝트 초기 세팅 (#7)

* init : 초기 설정

* chore : 글꼴 추가

* feat : 앱 반려동물 등록 시작화면 xml 추가

* feat : 반려동물 정보 등록창 xml 생성

* design: Dashboard 화면에 BottomNavigation 추가

- 화면 전환은 추후에 구현

* chore: .idea 폴더 삭제

chore: .idea 폴더 삭제

body
.idea 폴더는 intellij 세팅 폴더로 삭제

* docs : gitignore 수정 및 템플릿 추가

* docs: 3주차 리뷰 받고 싶은 내용 정리

---------

Co-authored-by: yb0x00 <yeongbeen0x00@gmail.com>
Co-authored-by: Lee Hyungi <63250720+geist9110@users.noreply.github.com>

fiveJinw committed:
Merge branch 'Master' into Develop

fiveJinw committed:
Merge pull request #8 from kakao-tech-campus-2nd-step3/Develop

[Week3] 프로젝트 초기 세팅 (#7)

LeeOhHyung commented on gradle/libs.versions.toml:
```diff
@@ -0,0 +1,26 @@
+[versions]
+agp = "8.3.1"
+kotlin = "1.9.0"
+coreKtx = "1.13.1"
+junit = "4.13.2"
```
신규 프로젝트를 진행하는것이니, 버전을 중간에 올리기보단 처음에 잡고 가는게 좋긴합니다.

[Kotlin 2.0.20 버전](https://kotlinlang.org/docs/whatsnew2020.html) 이 stable로 배포되어 있는데요. 조금 욕심을 내어서 이번 프로젝트에 아예 코틀린 2.0대로 개발을 해보시는건 어떠신가요?

LeeOhHyung replied:
그에따라 AGP 버전을 8.6~8.7로 올려야하긴 하겠지만, 아직 프로젝트 코드가 많지 않으니 어렵진 않을것입니다

fiveJinw replied:
좋은 리뷰 감사합니다! kotlin 2.0.20 버전으로 마이그레이션을 진행하였는데, android gradle plugin 버전도 추가로 변경해야하는지 궁금합니다!

LeeOhHyung replied:
@fiveJinw 빌드 및 실행이 정상적이라면 현재 AGP버전을 사용해도 괜찮습니다 
보통 사용하시는 안드로이드 스튜디오나 코틀린 버전에 맞는 AGP 버전을 설정해야 하는데, 이미 충분히 높은 버전을 사용하고 있다면 문제가 없을것 같네요
https://developer.android.com/build/releases/gradle-plugin?hl=ko#updating-gradle

LeeOhHyung commented on README.md:
```diff
@@ -1,2 +1,29 @@
+    Emulator를 사용하는 팀원과 실물 기기를 연결해 사용하는 팀원이 있어 화면상에 차이가 발생하는 부분을 최대한 줄이고 싶은 상황입니다.
+
+
+- Q. [BottomNavigationView ripple]
+
+
+<img src="https://github.com/user-attachments/assets/9829e702-71dc-49be-bc31-fd62de9692bf" width="300"/>
+
+   메뉴를 터치했을 때 보라색으로 이펙트가 있는 상황인데 이를 제거할 수 있는 방법을 여쭙고 싶습니다.
```
BottomNavagationView XML에 app:itemActiveIndicatorStyle="@android:color/transparent" 속성을 추가하면 없어질것 같습니다. 안된다면 코멘트에 남겨주세요 ㅠ

yb0x00 replied:
답변 감사합니다!
알려주신 것처럼 `app:itemActiveIndicatorStyle="@color/transparent"`
속성을 추가해 클릭했을 때 발생하는 보라색 배경을 제거 할 수 있었습니다. 😄

LeeOhHyung commented on README.md:
```diff
@@ -1,2 +1,29 @@
+   - 메인 화면 일부 구성
+
+#### 코치님께 여쭙고 싶은 내용
+- Q. [px -> dp/dpi 값 산정 어려움]
+
+  피그마로 기획안 화면을 바탕으로 Andorid Studio에서 화면을 그릴 때 DP를 계산해야하는 과정이 필요한데 이 부분에서 혼란을 느끼고 있습니다. 피그마로 기획한 화면에 대해 모든 값을 수치화해서 진행하는 것이 더 옯은 방향인지, 혹은 Pixel 4 화면을 바탕으로 개발자가 육안으로 확인하면서 화면을 그려도 괜찮은지 여쭙고 싶습니다.
+    Emulator를 사용하는 팀원과 실물 기기를 연결해 사용하는 팀원이 있어 화면상에 차이가 발생하는 부분을 최대한 줄이고 싶은 상황입니다.
```
아하. 피그마에서 아래 Layout영역에 있는 width / height가 px로 되어있으니 dp값으로 변환해서 쓰고 있으신것이죠?
그렇다면 걱정할거 없이 해당 수치를 그대로 넣으면 됩니다. 예를들어 피그마에 16px으로 적용되어 있다면 XML에 그대로 16dp으로 적용하면 되는것이죠. 

저도 디자인을 제공받을때 피그마를 주력으로 쓰지만, 웹/iOS에 디자인 기준으로 되어 있어서 그런지 px로 되어있긴한데 이걸 pxToDp로 매번 변환할 필요는 없고 바로 사용하시면 됩니다.

<p>
  <img src="https://github.com/user-attachments/assets/ade02209-2ee1-4efc-b0c4-5964f97b0def" width="250" />
</p>

yb0x00 replied:
해당 수치를 그대로 적용해도 되는 부분이었군요😮
답변 감사합니다 코치님:)

LeeOhHyung commented on app/src/main/res/layout/activity_dashboard.xml:
```diff
@@ -0,0 +1,80 @@
+        app:layout_constraintBottom_toTopOf="@id/home_frameLayout"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent">
+        <TextView
+            android:id="@+id/home_title"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text='@string/title'
+            android:textSize="24sp"
+            android:textStyle="bold"
+            android:textColor="@color/black"
+            android:fontFamily="@font/noto_sans_font"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            android:layout_marginTop="40dp"/>
```
activity_dashboard 이외에도 다른 화면들에서 TextView를 많이 사용하고 있더라구요
그때 마다 textSize, textStyle, fontFamily를 설정하면 코드가 늘어나기도 하고, 일괄 적용 및 사용처 추적에도 쉽지 않아 유지보수가 힘들어지는 특성이 있습니다.

그래서 Android Typography Design System / Android TextAppreance Theme 등등의 키워드로 검색을 해보시면, style.xml에 미리 필요한 프리셋을 정의해서 재활용하며 사용할 수 있습니다. Body, H1, Caption 등등 (웹개발을 해보셨다면 이미 익숙하실수있겟네요) 이름으로 추후에 `android:textAppearance="@style/..."` 으로 간단히 입력해서 사용할 수 있는 방법이 있습니다.

보통 이것도 프로젝트 초기에 미리 해두면 좋으니, 필요하다고 생각되신다면 지금 초기셋팅 단계에서 시도해보면 좋을것 같습니다.

- [1] https://github.com/HugoMatilla/Android-Design-System-and-Theming-Typography
- [2] https://medium.com/@rafael.gferreira98/what-is-android-textappearance-attribute-when-should-i-use-it-fc3e94a15c0e
- [3] https://m2.material.io/develop/android/theming/typography

fiveJinw replied:
좋은 리뷰 감사합니다! 디자인 초안을 바탕으로 자주 사용될법한 동일한 조건을 가진 텍스트들을 묶어서 style에 저장하였습니다. 혹시 잘못된 부분이 있는지 한번만 여쭈어보고 싶습니다!

LeeOhHyung commented on app/src/main/res/layout/activity_info_registration.xml:
```diff
@@ -0,0 +1,258 @@
+        android:textSize="12sp"
+        android:textStyle="bold"/>
+
+    <TextView
+        android:id="@+id/step_one_circle"
+        app:layout_constraintTop_toBottomOf="@+id/info_reg_subtitle_red"
+        app:layout_constraintEnd_toStartOf="@+id/dot_one_to_two"
+        android:layout_marginTop="29.6dp"
+        android:layout_marginEnd="8dp"
+        android:layout_width="23dp"
+        android:layout_height="23dp"
+        android:background="@drawable/red_circle"
+        android:text="1"
+        android:textSize="17sp"
+        android:gravity="center"
+        android:textStyle="bold"
+        android:textColor="@color/white"/>
```
약간의 팁(?) 겸 보정해서 봐야할게, 피그마에 나와있는 margin, width, height, padding 수치들이 홀수 이거나 소수점이라면 반올림 해서 넣거나 짝수화 시켜서 넣는것을 추천드립니다. 예를들면 29.6dp -> 30dp 이렇게 되는것이죠

디자이너 분이 디자인을 전달주실때, 다른 툴에서 피그마로 import시켰거나 혹은 iOS -> Android 디자인으로 변환하는 과정에서 수치들이 차이가 발생하더라구요.

LeeOhHyung reviewed (commented):
3주차 고생많으셨습니다.
참고사항에 있는 슈퍼용봉탕 페이지와 위클리도 확인했는데 초기에 체계를 갖추고 진행하시는것 같군요.

아직 본격적인 기능개발에 앞서 프로젝트 템플릿 및 초기세팅을 하는 느낌인것 같습니다.
남겨둔 코멘트 확인해주시고, 만약 수정사항을 반영하신다면 이 PR에 커밋 올려두시면 9월 23일 (월) 저녁에 확인 후 머지하도록 하겠습니다.

fiveJinw committed:
[Week3] 멘토님 리뷰 피드백 반영 (#13)

* feat : 앱 반려동물 등록 시작화면 xml 추가

* feat : 반려동물 정보 등록창 xml 생성

* design: Dashboard 화면에 BottomNavigation 추가

- 화면 전환은 추후에 구현

* chore: .idea 폴더 삭제

chore: .idea 폴더 삭제

body
.idea 폴더는 intellij 세팅 폴더로 삭제

* docs : gitignore 수정 및 템플릿 추가

* docs: 3주차 리뷰 받고 싶은 내용 정리

* style : xml dp 소수점 반올림

* docs: README 오타 수정

* chore: Gradle Migration 버전 1로 설정

- 자동 생성된 코드

* design: BottomNavigationView에서 아이콘 클릭 시 발생하는 배경 제거

- 피드백 반영 (https://github.com/kakao-tech-campus-2nd-step3/Team27_Android/pull/9)
- BottomNavigationView의 색상이 Emulator에서 투명도가 있는 흰색으로 보여 확인이 필요함

* fix: 원격 브랜치의 변경 사항 반영

- 코드 수정 전 원격 브랜치의 변경 사항을 반영하지 않아 PR에서 충돌 발생 (https://github.com/kakao-tech-campus-2nd-step3/Team27_Android/pull/10)
- 해결하기 위해 원격 브랜치의 변경 사항 반영

* style : style.xml 및 textAppearance 추가

* Revert "fix: 원격 브랜치의 변경 사항 반영"

This reverts commit 06e14e1867735c2abe619dc380e35d16f112dd8e.

* style : DashboardActivity 오타 수정

* chore : kotlin 2.0.20 버전으로 마이그레이션

---------

Co-authored-by: yb0x00 <yeongbeen0x00@gmail.com>
Co-authored-by: Lee Hyungi <63250720+geist9110@users.noreply.github.com>

fiveJinw committed:
Merge pull request #14 from kakao-tech-campus-2nd-step3/Develop

[Week3] 멘토님 리뷰 피드백 반영 (#13)

yb0x00 commented:
코치님 추가적인 질문이 있어 여기에 코멘트 남깁니다.
app/src/main/res/layout/activity_dashboard.xml
에서 BottomNavigationView 속성에
`android:background="@color/white"`가 있는데도 불구하고
에뮬레이터로 실행해보면
ConstraintLayout의 white와는 다른 좀 더 흐리고 탁한 white 색상이 보입니다.
제 에뮬레이터의 문제인지 아니면 설정이 필요한 다른 속성이 있는지 질문드리고 싶습니다.

- activity에서 색상을 white로 설정했을 때도 색상의 차이가 나타났습니다. (참고: https://stackoverflow.com/questions/66099082/android-kotlin-why-is-it-showing-white-background-from-bottomnavigationview-des)

LeeOhHyung commented:
@yb0x00 아하, 아래래 사진처럼 backgroundTint or / itemBackground 으로 해결가능하지 않을까 싶어요! 

<img width="549" alt="image" src="https://github.com/user-attachments/assets/573ec468-7274-45ab-9269-f1b0b2e2b509">

LeeOhHyung reviewed (approved):
Good! 

수정사항들도 빠르게 반영해주셨군요.
마지막 커밋들 확인했습니다. 감사합니다.
